### DISCUSSION TITLE
[Feature Request] Support Interactive Checkboxes in Markdown Tables

### DISCUSSION THREAD

**Discussion by BenjaminTJohnson** [community/community#61755] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

**Summary:** 

GitHub's Markdown supports many rich text features, including tables and interactive checkboxes. However, currently, interactive checkboxes are not supported within Markdown tables, only appearing as plaintext `- [ ]`. This feature request is to enable interactive checkboxes within Markdown tables to improve the efficiency and interactivity of task tracking within issues.

**Current Behavior:** 

When trying to insert an interactive checkbox within a Markdown table, it is rendered as plaintext. For example:

| Checkbox | Task  |
|:--------:|-------|
| - [ ]    | Task1 |
| - [ ]    | Task2 |

This code results in the `- [ ]` being displayed as plaintext, rather than an interactive checkbox.

**Desired Behavior:** 

We propose that GitHub Markdown should support interactive checkboxes within tables. Using the same code as above would be ideal, but certain open to other solutions that would accomplish the same effect.  

This should result in a table with interactive checkboxes, which can be ticked off directly within the rendered table similar to how checkboxes in lists currently work.  

**Benefits:**

This feature would bring a number of benefits:

1. **Efficiency:** It would allow users to track progress of tasks directly within tables, making it easier to visualize and manage task lists.
2. **Flexibility:** It would provide more options for users to format their content in a way that best suits their project's needs.
3. **Consistency:** Since interactive checkboxes are already supported outside of tables, this feature would make the Markdown behavior more consistent.

Thank you for considering this feature request.   We believe that it would significantly improve the user experience of using GitHub Markdown, and we look forward to any feedback or further discussion.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Rwlearn** [comment]
I am looking for this feature to be implemented for a long time. It would be very useful for us. Thank you
